assert/assertions.go

```diff
@@ -54,116 +55,28 @@ type Comparison func() (success bool)
 
 // ObjectsAreEqual determines if two objects are considered equal.
 //
-// This function does no assertion of any kind.
+// Deprecated: This function does no assertion of any kind. Use Equal instead.
```

Comment:
`ObjectsAreEqual` might have been used not as a mistake (vs assertions) but as a conveniency (to avoid `reflect.Equal` or a dependency on goolgle/go-cmp). (I admit to have done that myself) so the deprecation message should also handle that case.

Also, use [godoc links](https://go.dev/doc/comment#doclinks).

Reply:
I think in #1279 we established that testify is not a utility library, I think that includes our docs.

Godoc links added.

Reply:
Testify should not have been used as a utility library, but the fact is it has been used like that, at least in testing frameworks/libraries.

See https://github.com/search?q=assert.ObjectsAreEqual+language%3AGo+NOT+repo%3Astretchr%2Ftestify&type=code